pypeit/core/wavecal/wvutils.py

```diff
@@ -36,35 +37,55 @@ def parse_param(par, key, slit):
-    """
-    Given an input wavelength vector and an input good pixel mask, this code computes the delta_wave defined
-    to be wave_old[i]-wave_old[i-1]. The missing point at the end of the array is just appended to have an
-    equal size array.
+    r"""
+    Compute the change in wavelength per pixel.
 
-    Args:
-        wave (float `numpy.ndarray`_): shape = (nspec,)
-            Array of input wavelengths. Muse be a one dimensional array.
-        gpm (bool `numpy.ndarray`_): shape = (nspec)
-            Boolean good pixel mask defining where the wave_old are good.
-        frac_spec_med_filter (float, optional):
-            Fraction of the nspec to use to median filter the delta wave by to ensure that it is smooth. Deafult is
-            0.03. In other words, a running median filter will be applied with window equal to 0.03*nspec
+    Given an input wavelength vector and an input good pixel mask, the *raw*
+    change in wavelength is defined to be ``delta_wave[i] =
+    wave[i+1]-wave[i]``, with ``delta_wave[-1] = delta_wave[-2]`` to force
+    ``wave`` and ``delta_wave`` to have the same length.
+
+    The method imposes a smoothness on the change in wavelength by (1)
+    running a median filter over the raw values and (2) smoothing
+    ``delta_wave`` with a Gaussian kernel. The boxsize for the median filter
+    is set by ``frac_spec_med_filter``, and the :math:`\sigma` for the
+    Gaussian kernel is either a 10th of that boxsize or 3 pixels (whichever
+    is larger).
+
+    Parameters
+    ---------- 
+    wave : float `numpy.ndarray`_, shape = (nspec,)
+        Array of input wavelengths. Must be 1D.
+    gpm : bool `numpy.ndarray`_, shape = (nspec)
+        Boolean good-pixel mask defining where the ``wave_old`` values are
+        good.
+    frac_spec_med_filter : :obj:`float`, optional
+        Fraction of the length of the wavelength vector to use to median
+        filter the raw change in wavelength, used to impose a smoothness.
+        Default is 0.03, which means the boxsize for the running median
+        filter will be approximately ``0.03*nspec`` (forced to be an odd
+        number of pixels).
 
     Returns
     -------
     delta_wave (float `numpy.ndarray`_): shape = (nspec,)
-            Array of wavelength differences using np.diff, where the last pixel has been extrapolated
-
+        A smooth estimate for the change in wavelength for each pixel in the
+        input wavelength vector.
     """
+    # Check input
+    if wave.ndim != 1:
+        msgs.error('Input wavelength array must be 1D.')
 
-    if wave.ndim > 1:
-        msgs.error('This routine can only be run on one dimensional wavelength arrays')
     nspec = wave.size
     # This needs to be an odd number
     nspec_med_filter = 2*int(np.round(nspec*frac_spec_med_filter/2.0)) + 1
     delta_wave = np.zeros_like(wave)
+    # TODO: Why is the gpm included here? The sampling of the spectrum isn't
```

Comment:
Good question.  @jhennawi ?

Reply:
@kbwestfall and @profxj, this should probably be renamed to wave_gpm and not gpm. The wave_gpm is where wave > 1.0, which is how we identify good wavelengths in spectra. The point is that for echelle there are a bunch of pixels in the arrays with wavelength zero because our wavelength data model has the number of wavelengths equal to the number of pixels in the spectral direction. 

The sampling is not defined in locations where the wavelengths are bad, so this gpm filtering is the right thing to do, but it should be called wave_gpm and not gpm.

Reply:
Changed `gpm` to `wave_gpm`.